### PR TITLE
Limit the disk usage to avoid running out of disk capacity

### PR DESCRIPTION
**Pull request by morningman**
Set high watermark and flood stage of disk used capacity.
And forbid some operations if disk usage is too high.

ISSUE: #1701

### CODE REVIEW

File: be/src/olap/data_dir.cpp [hunk 1]
```diff
@@ -1057,4 +1058,35 @@ void DataDir::_remove_check_paths_no_lock(const std::set<std::string>& paths) {
     }
 }
 
+Status DataDir::update_capacity() {
+    try {
+        boost::filesystem::path path_name(_path);
```
Comment (chaoyli) [inline be/src/olap/data_dir.cpp]: std::filesystem

File: be/src/olap/compaction.cpp [hunk 2]
```diff
@@ -186,4 +186,17 @@ OLAPStatus Compaction::check_correctness(const Merger& merger) {
     return OLAP_SUCCESS;
 }
 
+OLAPStatus Compaction::check_disk_capacity() {
+    int64_t incoming_data_size = 0;
```
Comment (chaoyli) [inline be/src/olap/compaction.cpp]: only in SegmentWriter will be ok. All writers only go through the SegmentWriter.

File: be/src/olap/data_dir.cpp [hunk 3]
```diff
@@ -59,14 +59,14 @@ DataDir::DataDir(const std::string& path, int64_t capacity_bytes,
         TabletManager* tablet_manager, TxnManager* txn_manager)
         : _path(path),
         _capacity_bytes(capacity_bytes),
+        _available_bytes(0),
+        _disk_capacity_bytes(0),
```
Comment (chaoyli) [inline be/src/olap/data_dir.cpp]: _capacity_bytes and _disk_capacity_bytes are redundant ?

File: be/src/olap/schema_change.cpp [hunk 4]
```diff
@@ -1889,10 +1889,16 @@ OLAPStatus SchemaChangeHandler::_convert_historical_rowsets(const SchemaChangePa
         VLOG(10) << "begin to convert a history rowset. version="
                  << rs_reader->version().first << "-" << rs_reader->version().second;
 
+        // check disk capacity
+        int64_t tablet_size = rs_reader->rowset()->index_disk_size() + rs_reader->rowset()->data_disk_size();
+        if (sc_params.new_tablet->data_dir()->reach_capacity_limit(tablet_size)) {
```
Comment (chaoyli) [inline be/src/olap/schema_change.cpp]: Unify the entrance.

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/data_dir.cpp (hunk 3 above)
Comment (morningman) [reply be/src/olap/data_dir.cpp]: No, `_capacity_bytes` is user specified disk capacity, but it currently not used.
`_disk_capacity_bytes` is the capacity of the disk

File: be/src/olap/data_dir.cpp (hunk 1 above)
Comment (morningman) [reply be/src/olap/data_dir.cpp]: C++14 not support std::filesystem

Comment (morningman) [reply be/src/olap/data_dir.cpp]: C++14 not support std::filesystem

**Review by chaoyli — APPROVED** [review]

**Pull request merged**